api/catalog/api/views/image_views.py

```diff
@@ -132,7 +132,18 @@ def watermark(self, request, *_, **__):
         params.is_valid(raise_exception=True)
 
         image = self.get_object()
-        image_url = image.url
+
+        # initialising the image url to None by default
+        image_url = None
+
+        # checking if thumbnail url is available or not
+        # if yes , setting image_url to thumbnailurl
+        # if no , set image_url to its default url
+        if image.thumbnail is not None:
+            image_url = image.thumbnail
+        else:
+            image_url = image.url
```

Comment:
This can be simplified into 1 line. We could even drop the `is not None` part because an empty string in the thumbnail field is no good for us either.

```suggestion
        image_url = image.thumbnail if image.thumbnail else image.url
```

Reply:
Hey @dhruvkb, can you please review the changes suggested? Thank you very much.